apis/duck/v1alpha1/subscribable_types.go

```diff
@@ -17,65 +17,105 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	"github.com/knative/pkg/apis/duck"
 )
 
+const (
+	StatusKeyName = "subscribable"
+)
+
 // Subscribable is the schema for the subscribable portion of the payload
 type Subscribable struct {
-	// TODO(vaikas): Give me a schema!
-	Field string `json:"field,omitempty"`
+	// Channelable is a reference to the actual Channelable
+	// interface that provides the Subscription capabilities. This may
+	// point to object itself or to another object providing the
+	// actual implementation.
+	Channelable corev1.ObjectReference `json:"channelable,omitempty"`
 }
 
 // Implementations can verify that they implement Subscribable via:
-var _ = duck.VerifyType(&Topic{}, &Subscribable{})
+var _ = duck.VerifyType(&ChannelableRef{}, &Subscribable{})
 
 // Subscribable is an Implementable "duck type".
 var _ duck.Implementable = (*Subscribable)(nil)
 
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// Topic is a skeleton type wrapping Subscribable in the manner we expect
+// ChannelableRef is a skeleton type wrapping Subscribable in the manner we expect
 // resource writers defining compatible resources to embed it.  We will
 // typically use this type to deserialize Subscribable ObjectReferences and
 // access the Subscribable data.  This is not a real resource.
-type Topic struct {
+type ChannelableRef struct {
```

Comment:
The naming is confusing me without a greater understanding of the object model.

Reply:
yeah, original thinking was that having it called would be confusing. It's not, changed this to Subscription to mirror others.